ElectronClient/app/gui/NoteText.jsx

```diff
@@ -1235,81 +1235,92 @@ class NoteTextComponent extends React.Component {
-				} catch (error) {
-					console.error(error);
-					bridge().showErrorMessageBox(error.message);
-				}
-			} else if (target === 'printer') {
-				try {
-					await InteropServiceHelper.printNote(options.noteId, {
-						printBackground: true,
-						customCss: this.props.customCss,
-					});
-				} catch (error) {
-					console.error(error);
-					bridge().showErrorMessageBox(error.message);
-				}
+		await new Promise(resolve => {
+			setTimeout(resolve, 100);
+		});
 
-				// restoreSettings();
+		if (target === 'pdf') {
+			try {
+				const pdfData = await InteropServiceHelper.exportNoteToPdf(options.noteId, {
+					printBackground: true,
+					pageSize: Setting.value('export.pdfPageSize'),
+					landscape: Setting.value('export.pdfPageOrientation') === 'landscape',
+					customCss: this.props.customCss,
+				});
+				await shim.fsDriver().writeFile(options.path, pdfData, 'buffer');
+			} catch (error) {
+				console.error(error);
+				bridge().showErrorMessageBox(error.message);
+			}
+		} else if (target === 'printer') {
+			try {
+				await InteropServiceHelper.printNote(options.noteId, {
+					printBackground: true,
+					customCss: this.props.customCss,
+				});
+			} catch (error) {
+				console.error(error);
+				bridge().showErrorMessageBox(error.message);
 			}
-			this.isPrinting_ = false;
-		}, 100);
+		}
+		this.isPrinting_ = false;
 	}
 
 	async commandSavePdf(args) {
 		try {
-			if (!this.state.note && !args.noteId) throw new Error(_('Only one note can be exported to PDF at a time.'));
+			if (!this.state.note && !args.noteIds) throw new Error('No notes selected for pdf export');
 
-			const note = (!args.noteId ? this.state.note : await Note.load(args.noteId));
+			let noteIds = args.noteIds ? args.noteIds : [this.state.note.id];
 
-			const path = bridge().showSaveDialog({
-				filters: [{ name: _('PDF File'), extensions: ['pdf'] }],
-				defaultPath: safeFilename(note.title),
-			});
+			let path = null;
+			if (noteIds.length === 1) {
+				const note = await Note.load(noteIds[0]);
+				path = bridge().showSaveDialog({
+					filters: [{ name: _('PDF File'), extensions: ['pdf'] }],
+					defaultPath: `${safeFilename(note.title)}.pdf`,
+				});
+
+			} else {
+				path = bridge().showOpenDialog({
+					properties: ['openDirectory', 'createDirectory'],
+				});
+			}
 
 			if (!path) return;
 
-			await this.printTo_('pdf', { path: path, noteId: note.id });
+			let titles = [];
+			for (let i = 0; i < noteIds.length; i++) {
+				const note = await Note.load(noteIds[i]);
+				let pdfPath = path;
+
+				// find a unique name if multiple notes are saved
+				if (noteIds.length > 1) {
+					let title = note.title;
+					for (let j = 1; j < 1000; j++) {
+						if (!titles.includes(title)) {
+							break;
+						} else if (j === 1000) {
+							throw new Error('Cannot find unique filename');
+						}
+						title = `${note.title} (${j})`;
+					}
+					titles.push(title);
+
+					pdfPath += `/${title}.pdf`;
+				}
```

Comment:
There's a shim.fsDriver().findUniqueFilename - can it be used instead?

Reply:
Excellent, done.